src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -53,103 +68,844 @@ static ColliderJntSphInit D_809BF3F8 = {
+    Vec3f effectPos;
+
+    this->unk_196--;
+
+    if (this->unk_196 >= 0) {
+        Math_StepToF(&this->actor.world.pos.x, this->actor.home.pos.x + 263.0f, 263.0f / 21);
+
+        if (this->unk_196 < 14) {
+            this->actor.world.pos.y = sinf(this->unk_196 * (M_PI / 28)) * 200.0f + this->actor.home.pos.y;
+        } else {
+            this->actor.world.pos.y =
+                sinf((this->unk_196 - 7) * (M_PI / 14)) * 130.0f + (this->actor.home.pos.y + 70.0f);
+        }
+        if (this->unk_196 == 0) {
+            effectPos.x = this->actor.world.pos.x + 40.0f;
+            effectPos.y = this->actor.world.pos.y;
+            effectPos.z = this->actor.world.pos.z - 70.0f;
+            EffectSsGSplash_Spawn(globalCtx, &effectPos, NULL, NULL, 1, 2000);
+            effectPos.x = this->actor.world.pos.x - 40.0f;
+            EffectSsGSplash_Spawn(globalCtx, &effectPos, NULL, NULL, 1, 2000);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_LAND_WATER);
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOLON_LAND_BIG);
+            func_800AA000(0.0f, 0xFF, 0x14, 0x96);
+            return;
+        }
+    } else if (this->unk_196 < -1) {
+        this->actor.world.pos.y = this->actor.home.pos.y - (sinf((this->unk_196 + 1) * (M_PI / 10)) * 20.0f);
+        if (this->unk_196 == -10) {
+            func_809BD3AC(this);
+        }
+    }
+}
+
+void func_809BDAE8(EnBigokuta* this, GlobalContext* globalCtx) {
+    if (Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y + 0x4000, 0x400)) {
+        if (this->unk_196 != 0) {
+            this->unk_196--;
+        }
+        if (this->unk_196 == 0) {
+            func_809BCE3C(this);
+            this->actor.home.pos.y = this->actor.world.pos.y;
+            Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_ENEMY);
+            this->actor.params = 2;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_DAIOCTA_VOICE);
+            func_809BD3E0(this);
+        }
+    }
+}
+
+void func_809BDB90(EnBigokuta* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (this->unk_196 != 0) {
+        this->unk_196--;
+    }
+    if (this->unk_196 == 0) {
+        if (this->actor.params == 3) {
+            func_800F5ACC(0x38U);
+        }
+        func_809BD3F8(this);
+    }
+}
+
+void func_809BDC08(EnBigokuta* this, GlobalContext* globalCtx) {
+    Player* player;
+    s16 phi_v0;
+    s16 pad;
+    s16 phi_v1;
+    Vec3f sp28;
+
+    player = PLAYER;
+    SkelAnime_Update(&this->skelAnime);
+    if (Animation_OnFrame(&this->skelAnime, 0.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_OCTAROCK_BUBLE);
+    }
+
+    if (this->unk_196 < 0) {
+        this->actor.shape.rot.y += this->unk_194 * 0x200;
+        func_809BCE3C(this);
+        this->unk_196++;
+        if (this->unk_196 == 0) {
+            this->unk_196 = 350;
+        }
+        func_809BCF68(this, globalCtx);
+        return;
```

Comment:
I think there should be an else rather than an early return since it's no action setup or actor kill, but it would also be a big else so I'm not strongly opinionated